pkg/scheduler/objects/queue.go

```diff
@@ -1571,3 +1593,84 @@ func (sq *Queue) recalculatePriority() int32 {
 	sq.currentPriority = curr
 	return priorityValueByPolicy(sq.priorityPolicy, sq.priorityOffset, curr)
 }
+
+func (sq *Queue) UpdateApplicationPreemptionPriority(applicationID string, priority int32) {
+	if sq == nil || !sq.isLeaf {
+		return
+	}
+	value := sq.updateApplicationPreemptionPriorityInternal(applicationID, priority)
+	sq.parent.UpdateQueuePreemptionPriority(sq.Name, value)
+}
+
+func (sq *Queue) UpdateQueuePreemptionPriority(queueName string, priority int32) {
+	if sq == nil || sq.isLeaf {
+		return
+	}
+	value := sq.updateQueuePreemptionPriorityInternal(queueName, priority)
+	sq.parent.UpdateQueuePreemptionPriority(sq.Name, value)
+}
+
+func (sq *Queue) updateQueuePreemptionPriorityInternal(queueName string, priority int32) int32 {
+	sq.Lock()
+	defer sq.Unlock()
+
+	if _, ok := sq.children[queueName]; !ok {
+		log.Logger().Debug("Unknown queue", zap.String("queueName", queueName))
+		return sq.preemptionPriority
+	}
+
+	sq.childPreemptionPriorities[queueName] = priority
+	return sq.recalculatePreemptionPriority()
+}
+
+func (sq *Queue) updateApplicationPreemptionPriorityInternal(applicationID string, priority int32) int32 {
+	sq.Lock()
+	defer sq.Unlock()
+
+	if _, ok := sq.applications[applicationID]; !ok {
+		log.Logger().Debug("Unknown application", zap.String("applicationID", applicationID))
+		return sq.preemptionPriority
+	}
+
+	sq.appPreemptionPriorities[applicationID] = priority
+	return sq.recalculatePreemptionPriority()
+}
+
+func (sq *Queue) recalculatePreemptionPriority() int32 {
+	var items *map[string]int32
```

Comment:
I am not sure that I understand this indirection. If I say:
```
items = sq.appPreemptionPriorities
```
both will point to the same data structure it will not copy the data.

Reply:
I guess the idea here is that by using a pointer it's a tad cheaper to iterate over the map because we avoid copying.

Reply:
There is no copy when you do the assign, see this example code:
https://go.dev/play/p/qcJ9P-4ZCdB
Both point to one and the same data structure

Reply:
Oh, that's great. Makes sense to change then.